### PR TITLE
Refactoring texteditor component

### PR DESCRIPTION
**Pull request by henrywangzh**

### CODE REVIEW

File: src/components/TextEditor/components/TextEditor.js [hunk 1]
```diff
@@ -29,136 +29,152 @@ if (typeof window !== 'undefined' && typeof window.navigator !== 'undefined') {
...
+    if (dirty) {
       ev.returnValue = '';
     }
     return ev;
   };
 
-  setCodeMirrorInstance = (codeMirrorInstance) => {
-    this.setState({ codeMirrorInstance });
-  };
+  // fix this
```
Comment (lumisphere902) [inline src/components/TextEditor/components/TextEditor.js:112]: Something needs to be fixed or is this old?

File: src/components/TextEditor/components/TextEditor.js [hunk 2]
```diff
@@ -29,136 +29,152 @@ if (typeof window !== 'undefined' && typeof window.navigator !== 'undefined') {
...
+  const setCurrentLineManual = (cm) => {
     const { line } = cm.getCursor();
     if (codeMirrorInstance) {
       // removeLineClass removes the back highlight style from the last selected line
       codeMirrorInstance.removeLineClass(currentLine, 'wrap', 'selected-line');
       // addLineClass adds the style to the newly selected line
       codeMirrorInstance.addLineClass(line, 'wrap', 'selected-line');
     }
-    this.setState({ currentLine: line });
+    setCurrentLineManual(line);
```
Comment (lumisphere902) [inline src/components/TextEditor/components/TextEditor.js:130]: function calls itself?

**Review by lumisphere902 — CHANGES_REQUESTED** [review]
Just a few small issues. Also, there are some merge conflicts; you should fix those and update w/ latest master

File: src/components/TextEditor/components/TextEditor.js (hunk 2 above)
Comment (henrywangzh) [reply src/components/TextEditor/components/TextEditor.js:130]: whoops, will fix both these issues rn

Comment (henrywangzh) [reply src/components/TextEditor/components/TextEditor.js]: hmm I'm a bit confused, it says I'm up to date when I do git pull and yet the code I have isn't the same as the master.

Comment (lumisphere902) [reply src/components/TextEditor/components/TextEditor.js]: git merge master

Comment (henrywangzh) [reply src/components/TextEditor/components/TextEditor.js]: I think something is definitely wrong now

**Comment by lumisphere902** [comment]
OK you kind of just...reverted all the master changes. I can help you resolve merge conflicts and stuff sometime this weekend.

File: src/components/TextEditor/components/TextEditor.js (hunk 2 above)
Comment (henrywangzh) [reply src/components/TextEditor/components/TextEditor.js]: Ok I have reverted the changes to before the merge, but every time I try to git merge master, it changes 50 files which I'm slightly concerned about.

Comment (reginawang3495) [reply src/components/TextEditor/components/TextEditor.js]: Honestly, I’d just take the git diff and apply it to another branch thats been freshly branched off of master 

It probably isn’t worth your time to fix up!

**Comment by lumisphere902** [comment]
PR moved to #709

**Pull request closed without merging**